### PR TITLE
DDF-4192 Add validation to search forms

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Validate search forms (adhoc, basic, advanced) before a search/save is run
#### Who is reviewing it? 
@Bdthomson @hayleynorton @alanchampion 

#### Select relevant component teams: 

@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining @andrewkfiedler 

#### How should this be tested?

- build/run
- currently validation should always be true, so verify that functionality of all search forms (adhoc, advanced, basic) remains the same.
#### What are the relevant tickets?
[DDF-4192](https://codice.atlassian.net/browse/DDF-4192)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bdeining** [comment]
@andrewzimmer is it possible to post some screenshots?

**Comment by bdeining** [comment]
screen shot 2018-10-01 at 3 48 39 pm

File: ui/packages/catalog-ui-search/src/main/webapp/component/announcement/CommonMessages.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const InvalidSearchFormMessage = {
+  title: 'Validation Issues: Search Form cannot be run.',
+  message: 'Please verify your configurations and attempt to search again.',
```
[the remark this answers is not in the document]
Comment (Lambeaux) [reply ui/packages/catalog-ui-search/src/main/webapp/component/announcement/CommonMessages.tsx]: ❓ How about
```
Please verify your search criteria and settings before attempting to search again
```

File: ui/packages/catalog-ui-search/src/main/webapp/component/query-add/query-add.view.js [hunk 2]
```diff
@@ -201,6 +202,10 @@ module.exports = Marionette.LayoutView.extend({
     this.queryContent.currentView.setDefaultTitle()
   },
   saveRun: function() {
+    if (!this.queryContent.currentView.isValid()) {
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/query-add/query-add.view.js]: ❓ Do we need it on the `save` method here or is that vestigial?

**Comment by Bdthomson** [comment]
build now

**Comment by bdeining** [comment]
built and verified

**Pull request merged**